cygnus-arcgis/src/main/java/com/telefonica/iot/cygnus/sinks/NGSIArcgisFeatureTableSink.java

```diff
@@ -0,0 +1,789 @@
+        } // try catch
+
+        super.start();
+    } // start
+
+    @Override
+    public void stop() {
+        super.stop();
+        if (arcgisPersistenceBackend != null) {
+            for (Map.Entry<String, NGSIArcgisFeatureTable> backend : arcgisPersistenceBackend.entrySet()) {
+                backend.getValue().flushBatch();
+            }
+            arcgisPersistenceBackend.clear();
+        }
+    } // stop
+
+    /**
+     * Initialices a lazy singleton to share among instances on JVM.
+     */
+    private static synchronized void createPersistenceBackend() {
+        if (arcgisPersistenceBackend == null) {
+            arcgisPersistenceBackend = new ConcurrentHashMap<String, NGSIArcgisFeatureTable>();
+        }
+    }
+
+    @Override
+    void persistBatch(NGSIBatch batch)
+            throws CygnusBadConfiguration, CygnusPersistenceError, CygnusRuntimeError, CygnusBadContextData {
+        if (batch == null) {
+            LOGGER.debug("[" + this.getName() + "] Null batch, nothing to do");
+            return;
+        } // if
+
+        try {
+            // Iterate on the destinations
+            batch.startIterator();
+
+            while (batch.hasNext()) {
+                String destination = batch.getNextDestination();
+                LOGGER.debug(
+                        "[" + this.getName() + "] Processing sub-batch regarding the " + destination + " destination");
+
+                // Get the events within the current sub-batch
+                ArrayList<NGSIEvent> events = batch.getNextEvents();
+
+                // Get an aggregator for this destination and initialize it
+                NGSIArcgisAggregator aggregator = new NGSIArcgisAggregator(getrAcgisServicesUrl(), enableNameMappings);
+                // aggregator.setService(events.get(0).getServiceForNaming(enableNameMappings));
+                // aggregator.setServicePathForData(events.get(0).getServicePathForData());
+                // aggregator.setServicePathForNaming(events.get(0).getServicePathForNaming(enableGrouping,
+                // enableNameMappings));
+                // aggregator.setEntityForNaming(events.get(0).getEntityForNaming(enableGrouping, enableNameMappings,
+                // enableEncoding));
+                // aggregator.setEntityType(events.get(0).getEntityTypeForNaming(enableGrouping, enableNameMappings));
+                // aggregator.setAttribute(events.get(0).getAttributeForNaming(enableNameMappings));
+                // aggregator.setDbName(buildDbName(aggregator.getService()));
+                // aggregator.setTableName(buildTableName(aggregator.getServicePathForNaming(),
+                // aggregator.getEntityForNaming(), aggregator.getEntityType(), aggregator.getAttribute()));
+                // aggregator.setAttrNativeTypes(attrNativeTypes);
+                // aggregator.setAttrMetadataStore(false);
+                aggregator.initialize(events.get(0));
```

Comment:
Is this call necessary?

Reply:
Removed 
https://github.com/Stdri/fiware-cygnus/commit/1941e81055e2d76d0b345879d0e68e244f0d241b